speculation-rules.bs

```diff
@@ -204,8 +207,7 @@ Periodically, for any [=document=] |document|, the user agent may [=queue a glob
     1. [=list/For each=] |rule| of |ruleSet|'s [=speculation rule set/prefetch rules=]:
       1. Let |requiresAnonymousClientIPWhenCrossOrigin| be true if |rule|'s [=speculation rule/requirements=] [=set/contains=] "`anonymous-client-ip-when-cross-origin`", and false otherwise.
       1. [=list/For each=] |url| of |rule|'s [=speculation rule/URLs=]:
-        1. The user agent may prefetch |url| given |requiresAnonymousClientIPWhenCrossOrigin|.
-           <p class="issue">TODO: expand this to actually elaborate on how prefetch works, once initiated, and to incorporate the |requiresAnonymousClientIPWhenCrossOrigin| flag. We may wish to include language about when the UA should deduplicate requests.
+        1. The user agent may [=prefetch=] |url| given |document|, |url| and "`strict-origin-when-cross-origin`". If |requiresAnonymousClientIPWhenCrossOrigin| is true, it must not do so unless it can obscure the user's client IP address for any request for a URL whose [=url/origin=] is not [=same origin=] to |document|'s [=Document/origin=].
```

Comment:
I would think that long term we would want to flow this flag down into https://fetch.spec.whatwg.org/#http-network-or-cache-fetch , probably step 9.5. So leaving a TODO might be a good idea, or maybe passing the boolean to prefetch and moving the TODO there.

Reply:
Hmm, all the way to https://fetch.spec.whatwg.org/#concept-connection-obtain? Even if we do plumb it there in the future it will probably remain a handwave there until/unless the mechanism for anonymizing the client IP becomes universal (e.g., everyone implements MASQUE proxies or whatever). For the time being this being implementation-defined seems appropriate.

Still, happy to leave a TODO here.

Reply:
Yeah. Making it clear that it normatively impacts the connection-obtaining part and nothing else would be ideal. I wouldn't expect anything more than a handwave there.

Right now it's essentially monkey-patching "obtain a connection" from a distance, which is not great.

In other words I think it's fine for there to be implementation-defined behavior at that level, we should just avoid the monkeypatching approach.

Anyway, the issue text you have below looks good to me for now.